test/dialect/postgresql/test_dialect.py

```diff
@@ -1210,3 +1212,67 @@ def test_get_dialect(self):
     def test_async_version(self):
         e = create_engine("postgresql+psycopg_async://")
         is_true(isinstance(e.dialect, psycopg_dialect.PGDialectAsync_psycopg))
+
+
+class TestRange(fixtures.TestBase):
+    __only_on__ = "postgresql"
+    __backend__ = True
+
+    @testing.combinations(0, 1, 2, 4, 5, argnames="v")
+    @testing.combinations(
+        "empty",
+        "(,)",
+        "(,4]",
+        "[1,)",
+        "[1,4)",
+        "[1,4]",
+        "(1,4]",
+        "(1,4)",
+        argnames="rrepr",
+    )
```

Comment:
**mike bayer** (zzzeek) wrote:

so here I want to use literal(), like this:

    connection.scalar(
        select(literal(r, INT4RANGE).contains(v))
    )

will that work?

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4151